core/hotspot/rule_manager.go

```diff
@@ -431,6 +431,9 @@ func IsValidRule(rule *Rule) error {
 	if rule.MetricType == QPS && rule.DurationInSec <= 0 {
 		return errors.New("invalid duration")
 	}
+	if rule.ParamIndex > 0 && rule.ParamKey != "" {
```

Comment:
it should be `rule.ParamIndex >= 0`

Reply:
In order to avoid the problem of paramInde default value 0, 
adjust the order of extracting parameters, 
that is, give priority to paramKey. 
The judgment conditions here will not be adjusted temporarily.